### PR TITLE
fix(compogen): fix wrong anchors and missing object descriptions

### PR DESCRIPTION
**Pull request by pinglin**
Because

- the auto-gen `oneOf` objects mess up the TOC of the documentation website due to wrong anchor syntax
- the object descriptions are missing
- `make build-doc` should use local codebase to build and install or otherwise the local change of `tools/compogen` can't be applied to auto-gen the latest component `README.mdx`.

This commit

- fix the issues mentioned above
- remove redundant empty lines in the readme template
- implement the precedence logic of task and `oneOf` object `description` and `instillShortDescription` 
  - `description` and `instillShortDescription` are used by the console to provide a description of the task/object in the component. If `instillShortDescription` does not exist, it will be the same as `description`.

### CODE REVIEW

File: tools/compogen/pkg/gen/readme.go [hunk 1]
```diff
@@ -536,33 +542,72 @@ func enumValues(enum []string) string {
 	return result
 }
 
-func findConstantValue(option objectSchema) string {
+func findConstantValue(option objectSchema, taskOrString interface{}) string {
```
[the remark this answers is not in the document]
Comment (pinglin) [reply tools/compogen/pkg/gen/readme.go]: I renamed both:
- `findConstantValue` to `insertHeaderByConstValue` 
- `headerWithID` to `insertHeaderByObjectKey`

**Review by chuang8511 — APPROVED** [review]
lgtm. Thanks.

File: tools/compogen/pkg/gen/resources/templates/readme.mdx.tmpl [hunk 2]
```diff
@@ -7,146 +7,157 @@ description: "Learn about how to set up a VDP {{ .Title }} component https://git
 
 The {{ .Title }} component is {{ .ComponentType.IndefiniteArticle }} {{ .ComponentType }} component that allows users to {{ firstToLower .Description }}.
 It can carry out the following tasks:
-{{ range .Tasks }}
-- [{{ .Title }}](#{{ asAnchor .Title}}){{ end }}
+{{- range .Tasks }}
```
Comment (jvallesm) [inline tools/compogen/pkg/gen/resources/templates/readme.mdx.tmpl:10]: [TIL](https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax#trim-markers) 🥇 This is gold, makes everything much more readable

**Review by jvallesm — COMMENTED** [review]
Thans for addressing this 🙇

**Pull request merged**